### ISSUE TITLE
PHP 7.4 errors in dependencies

### ISSUE THREAD

**Issue by faishal** [humanmade/altis-cloud#314]
We found the following errors after running PHP 7.4 compatibility

### vendor/stuttter/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php: 1 errors
-  [1335] : Since PHP 7.0, functions inspecting arguments, like debug_backtrace(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter "$query" was changed on line 1223. - PHPCompatibility.FunctionUse.ArgumentFunctionsReportCurrentValue.Changed
  
### vendor/humanmade/s3-uploads/lib/getid3/module.audio-video.quicktime.php: 1 errors
- [120] : list() will assign variable from left-to-right since PHP 7.0. Ensure all variables in list() are unique to prevent unexpected results. - PHPCompatibility.Lists.AssignmentOrder.Affected

**Comment by roborourke** [comment]
Altis v4 and up all use LudicrousDB at version 5 from the HM fork - the direct reference to `debug_backtrace` has been removed so nothing needs doing there.

I did however find the following issues running PHP compatilbity:

```
FILE: /Users/rob/projects/altis-dev/extra-packages/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php
----------------------------------------------------------------------------------------------------------------------------------
FOUND 21 ERRORS AFFECTING 21 LINES
----------------------------------------------------------------------------------------------------------------------------------
  809 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
  810 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
  952 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
  954 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
  981 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
  995 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1024 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1049 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1077 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1079 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1137 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1208 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1427 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1445 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1453 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1471 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1523 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1561 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1645 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 1715 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
 2099 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
      |       | (PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved)
----------------------------------------------------------------------------------------------------------------------------------


```

**Comment by roborourke** [comment]
Never mind - guess they were removed as it's just a backwards compatiblity thing, mysqli is used where available.

**Comment by roborourke** [comment]
So, nothing to do with ludicrousdb.

Regarding S3 uploads the bundled version of `getID3` is now behind the version as far back as WordPress 5.4 which is as far back as we support in Altis v4.

We need to either update the version or remove it from S3 Uploads and tag a new 2.3.0 release. Thoughts on that @joehoyle ?

**Comment by roborourke** [comment]
PR for potential 2.3.0 release of S3 Uploads here https://github.com/humanmade/S3-Uploads/pull/500

**Comment by roborourke** [comment]
Fixed in #344
